app/transforms/date-only.js

```diff
@@ -0,0 +1,13 @@
+import moment from 'moment';
+import DS from "ember-data";
+
+export default DS.Transform.extend({
+  serialize: function(value) {
+    return value ? value.toJSON() : null;
+  },
+
+  deserialize: function(value) {
+    return moment(value);
```

Comment:
I think we are close
> > A date, or partial date (e.g. just year or year + month) as used in human communication. There is no time zone.
union of xs:date, xs:gYearMonth, xs:gYear	A JSON string - a union of xs:date, xs:gYearMonth, xs:gYear
Regex: -?[0-9]{4}(-(0[1-9]|1[0-2])(-(0[0-9]|[1-2][0-9]|3[0-1]))?)?

To prevent deprecation warnings e.g

> Deprecation warning: value provided is not in a recognized RFC2822 or ISO format. moment construction falls back to js Date(), which is not reliable across all browsers and versions. Non RFC2822/ISO date formats are discouraged and will be removed in an upcoming major release. Please refer to http://momentjs.com/guides/#/warnings/js-date/ for more info.

from moment for a partial date e.g `2016`, we might need to add a default format. Given the above FHIR reference, I would recommend
```
moment(value, 'YYYY-MM-DD')
```

This should handle year only, year-month combination, and year-month-date combination